### DISCUSSION TITLE
Obviously, MAMP PRO on Mac is dead, what are you using instead for your local environment?

### DISCUSSION THREAD

**Discussion by Chacapamac** [templaza/astroid-framework#654] [Q&A]
Some contender?
XAMPP — AMPPS — Devilbox — DDEV — Fenix Web Server — Lando, BAMP, and Clamp — Laragon....???

**What you guys try and prefer?**

**Comment by sonvnn** [comment]
Xampp always is the best

**Comment by vtheod** [comment]
WampServer for local development, but I prefer developing on a VPS

**Reply by Chacapamac** [reply]
Thanks @vtheod 
**Can you do that on Mac?**
Also I have a older System Software (High Sierra)?

**Reply by vtheod** [reply]
Sorry, I'm not working on Mac but with a fast search I found this: https://formac.informer.com/wampserver

**Comment by Giorgi625** [comment]
I use local servers only to install older versions of my websites if I need to see how I did something in previous versions and compare to actual. I use Xampp two different versions one with PHP 7 for very old J3 projects and another with PHP 8 to check J4 and J5 backups.

Current projects I build on a hosting always.

**Reply by Chacapamac** [reply]
Thanks @Giorgi625  
> I use Xampp

I just install **XAMPP OSX 8.2.4-0** on my Mac High Sierra, everything seem Ok, so far... 

- PHP 8.2.4
- Apache 2.4.56
- MariaDB 10.4.32	
- phpMyAdmin 5.2.1
- OpenSSL 1.1.1t

**Testing Right Now!** 😀
So far the system seem to be relatively easy to use...

**Comment by ghost** [comment]
Mamp pro is still available for the Mac.

 https://www.mamp.info/de/store/

Otherwise, [Bearsampp](https://bearsampp.com/) is very similar but unfortunately not available for the Mac. But for me Bearsampp is the best for multiple virtual hosts and quickly changing the database version or PHP version.

**Comment by Chacapamac** [comment]
@vsmo1 
`Mamp pro is still available for the Mac.`
Yes, but the MYSQL onboard the last Mamp Pro 6 version is only **5.7** — Joomla 5, as MINIMUM ask for **MySQL 8.0.13** or MariaDB 10.4.0 and PHP 7.2.5.

**Comment by ghost** [comment]
Ok, then I misunderstood, I understood it as if Mamp Pro had been discontinued.

I faced the same problem a few weeks ago. I have Mamp Pro 6 for Windows but it was no longer sufficient for Joomla 5.

I was looking for a replacement that would allow me to edit multiple virtual hosts simultaneously and change PHP or database versions with just a few clicks.

I found Bearsampp and I am totally happy with it. I can easily create multiple virtual hosts and switch between them at will. With one click I can change the PHP version, or the database version (MariaDb or MySql).

But unfortunately Bearsampp is only available for Windows. So it won't help you if you use a Mac.

**Reply by Chacapamac** [reply]
Thank You for your input @vsmo1

**Comment by Chacapamac** [comment]
**OK, XAMPP IS ON!**

My last hurtle before installing Joomla 5 is a problem the Joomla “Recommended Settings” — I cannot find the way to turn the setting “intl Available” On??????

![Screen Shot 2024-04-27 at 10 49 29 AM](https://github.com/templaza/astroid-framework/assets/1907276/f8e44900-e126-4fdb-89b8-c980ef5a970f)

Anybody have that problem with the **XAMPP 8.2.4-0 on Mac**?

**Comment by Chacapamac** [comment]
I still have my problem with the Xampp server on my **Mac** when passing to Joomla 5, — Look at my previous post?
Joomla 5 Pre-check

Bing give me that bull to fix it?????

> If the `intl` extension still does not load, you may need to install it using `pecl`. Here are the steps:
> 
> 1. Install `icu4c` if it's not already installed:
>    ```
>    brew install icu4c
>    ```
> 2. Install the `intl` extension via `pecl`:
>    ```
>    sudo pecl install intl
>    ```
> 3. Add the `intl` extension to your `php.ini`:
>    ```
>    extension=intl.so
>    ```
> 4. Restart Apache.
> 
> Make sure that the PHP binary used in the terminal is the one from XAMPP. You can verify this by typing `which php` in the terminal. It should point to the XAMPP directory.
> 
> If you continue to face issues, it might be helpful to check the PHP error log for any specific error messages that can guide us to the solution. The error log can typically be found in `/Applications/XAMPP/xamppfiles/logs/` directory. Feel free to share any error messages you find there, and I'll be glad to assist further!

**Comment by iorbita** [comment]
MAMP Pro 7 is out :) https://www.mamp.info/en/mac/

**Reply by Chacapamac** [reply]
Great News 👍

**Comment by paurray** [comment]
@iorbita thanks for this. Definitely did not see this coming. Also on a Mac here and if/when MAMP works it is fantastic!!!

**Reply by iorbita** [reply]
@paurray I don't get the impression that it was announced with much fanfare either... the important thing is that the project has not been abandoned ✌️

**Comment by Chacapamac** [comment]
This is excellent news, back to MAMP 🏆

**Reply by Chacapamac** [reply]
Well not so fast, need Operating system: macOS Big Sur 11 or higher. Need to upgrade my IMac 2011. 😞

**Comment by ByteCrusher99** [comment]
ServBay https://www.servbay.com, portable and convenient, updates quickly

**Comment by dangelion** [comment]
I'm on Mac M4 Pro and Sequoia 15.3.1 (24D70), everytime I add a new host on MAMP PRO 7.2.2 Mac crashes (it reboots itself). Anyone with this annoying issue? That said I always hated Mamp, shit app, but it's the only one for Mac.

**Reply by johanguse** [reply]
Are you still having issues with 7.2.4? Sequoia is now at 15.4.1

**Comment by paurray** [comment]
@dangelion Understand the MAMP frustrations. I have a roughly similar setup, and things have been stable for 6 months +/-

**Comment by lamriabdellah0** [comment]
You're right—MAMP PRO on Mac seems to be pretty much on its way out. For a local environment, I’ve been really impressed with ServBay. It’s a solid choice, especially when you need something that’s lightweight, easy to set up, and provides a more robust and modern alternative. ServBay runs smoothly, offers great flexibility, and it’s been a total lifesaver for my projects, especially when compared to MAMP. If you're looking for a reliable and efficient local server environment, it’s definitely worth checking out.

**Comment by KeepCoolCH** [comment]
I moved from MAMP to my own App called [DockAMP](https://github.com/KeepCoolCH/DockAMP), which uses Docker Containers on the Backend. I wanted a simple UI for setting up Servers with no need to create Docker Compose Files. Maybe that's a solution for you to.
